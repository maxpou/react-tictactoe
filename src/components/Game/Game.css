.game {
  display: flex;
  flex-direction: row;
}

.game-info {
  margin-left: 20px;
}

.game-info-current {
  color: #61dafb;
}