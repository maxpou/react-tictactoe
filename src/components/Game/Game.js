import React from 'react'
import Board from '../Board/Board'
import DevTools from 'mobx-react-devtools'
import { observer } from 'mobx-react'
import './Game.css'

class Game extends React.Component {
  render () {
    const {game} = this.props

    const moves = game.history.map((step, move) => {
      const desc = move ? 'Go to move #' + move : 'Go to game start'
      return (
        <li key={move} className={move === game.stepNumber ? 'game-info-current' : ''}>
          <button onClick={() => game.jumpTo(move)}>{desc}</button>
        </li>
      )
    })

    return (
      <div className='game'>
        <div className='game-board'>
          <Board
            squares={game.current.squares}
            onClick={(i) => game.play(i)}
          />
        </div>
        <div className='game-info'>
          <div>{game.getStatus(game.current.squares)}</div>
          <ol>{moves}</ol>
        </div>
        <DevTools />
      </div>
    )
  }
}

export default observer(Game)
