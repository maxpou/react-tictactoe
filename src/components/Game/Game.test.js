import React from 'react'
import { shallow, mount } from 'enzyme'
import Game from './Game'

describe('Game.js', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Game />)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Same player should not play 2 times in a row', () => {
    const wrapper = mount(<Game />)
    const firstPlayer = wrapper.find('div.game-info').children().first().text()
    expect(firstPlayer).toEqual('Next player: X')

    const button = wrapper.find('button.square').first()
    button.simulate('click')
    const secondPlayer = wrapper.find('div.game-info').children().first().text()
    expect(secondPlayer).toEqual('Next player: O')
  })

  it('should declare a winner', () => {
    const wrapper = mount(<Game />)

    const round1 = wrapper.find('button.square').at(0)
    round1.simulate('click')
    const round2 = wrapper.find('button.square').at(1)
    round2.simulate('click')
    const round3 = wrapper.find('button.square').at(3)
    round3.simulate('click')
    const round4 = wrapper.find('button.square').at(2)
    round4.simulate('click')
    const round5 = wrapper.find('button.square').at(6)
    round5.simulate('click')

    const winnerDeclaration = wrapper.find('.game-info').children().first().text()
    console.log(winnerDeclaration)
    expect(winnerDeclaration).toEqual('Winner: X')
  })
})
