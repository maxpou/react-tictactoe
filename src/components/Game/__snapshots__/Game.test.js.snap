// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game.js renders without crashing 1`] = `"<div class=\\"game\\"><div class=\\"game-board\\"><div><div class=\\"board-row\\"><button class=\\"square\\"></button><button class=\\"square\\"></button><button class=\\"square\\"></button></div><div class=\\"board-row\\"><button class=\\"square\\"></button><button class=\\"square\\"></button><button class=\\"square\\"></button></div><div class=\\"board-row\\"><button class=\\"square\\"></button><button class=\\"square\\"></button><button class=\\"square\\"></button></div></div></div><div class=\\"game-info\\"><div>Next player: X</div><ol><li class=\\"game-info-current\\"><button>Go to game start</button></li></ol></div></div>"`;
